### ISSUE TITLE
Add gene_symbols argument to scanpy.pl.rank_genes_groups_matrixplot

### ISSUE THREAD

**Issue by ShanSabri** [scverse/scanpy#1427]
- [x] Additional function parameters / changed functionality / changed defaults?

Would it be possible to add the `gene_symbols=` argument to `scanpy.pl.rank_genes_groups_matrixplot`, just as done with `sc.pl.umap`? 

I'd like to take ENSG IDs all the way through the analysis (as `var.index`) and am having a hard time trying to unique the GeneIDs. I'm currently achieving this by: 

```
def make_genes_unique(adata):
    '''
    - makes gene names unique by appending a number string to each duplicate gene ID
    '''
    v = adata.var.groupby(adata.var.Name).cumcount().astype(str).str.replace('0', '').values
    adata.var.Name = adata.var.Name.values + ['-'+i if i !='' else i for i in v]
    return(adata)
```
 But there are still issues that come up along the way. Happy to discuss further.

**Comment by fidelram** [comment]
Certainly we need to do this not only for matrix plot but for dotplot. I will take a look.

As a workaround you can always do:

`adata.var = adata.var.reset_index().set_index('gene_name')`

**Comment by ivirshup** [comment]
Implemented in #1529
